cpp/src/arrow/flight/sql/odbc/flight_sql/flight_sql_connection.cc

```diff
@@ -62,24 +62,32 @@ using driver::odbcabstraction::DriverException;
 using driver::odbcabstraction::OdbcVersion;
 using driver::odbcabstraction::Statement;
 
-const char FlightSqlConnection::DSN[] = "dsn";
-const char FlightSqlConnection::DRIVER[] = "driver";
-const char FlightSqlConnection::HOST[] = "host";
-const char FlightSqlConnection::PORT[] = "port";
-const char FlightSqlConnection::USER[] = "user";
-const char FlightSqlConnection::USER_ID[] = "user id";
-const char FlightSqlConnection::UID[] = "uid";
-const char FlightSqlConnection::PASSWORD[] = "password";
-const char FlightSqlConnection::PWD[] = "pwd";
-const char FlightSqlConnection::TOKEN[] = "token";
-const char FlightSqlConnection::USE_ENCRYPTION[] = "useEncryption";
-const char FlightSqlConnection::DISABLE_CERTIFICATE_VERIFICATION[] =
+// clang-format off
+const std::string FlightSqlConnection::DSN = "dsn";            // NOLINT(runtime/string)
+const std::string FlightSqlConnection::DRIVER = "driver";      // NOLINT(runtime/string)
+const std::string FlightSqlConnection::HOST = "host";          // NOLINT(runtime/string)
+const std::string FlightSqlConnection::PORT = "port";          // NOLINT(runtime/string)
+const std::string FlightSqlConnection::USER = "user";          // NOLINT(runtime/string)
+const std::string FlightSqlConnection::USER_ID = "user id";    // NOLINT(runtime/string)
+const std::string FlightSqlConnection::UID = "uid";            // NOLINT(runtime/string)
+const std::string FlightSqlConnection::PASSWORD = "password";  // NOLINT(runtime/string)
+const std::string FlightSqlConnection::PWD = "pwd";            // NOLINT(runtime/string)
+const std::string FlightSqlConnection::TOKEN = "token";        // NOLINT(runtime/string)
+const std::string FlightSqlConnection::USE_ENCRYPTION =        // NOLINT(runtime/string)
+    "useEncryption";
+const std::string FlightSqlConnection::DISABLE_CERTIFICATE_VERIFICATION = // NOLINT
```

Comment:
Note to myself: After this PR is merged, I will make comment on https://github.com/apache/arrow/pull/40939:

Using `std::string` is more suitable in the code base. Is usage of `NOLINT` ok here?